### ISSUE TITLE
Domain shows too many items (at least with composed chart)

### ISSUE THREAD

**Issue by jukkahuuskonen** [recharts/recharts#3085]
I'm using a Composed chart with bars and area. The X-axel is time (milliseconds after unix zero date).

Sometimes when the time-value is just slightly smaller then domain lower limit, it will still show the bar on chart (same for the upper limit). This is somehow inconsistent, for example in here there is much smaller difference between domain upper limit and first item to be cut out than on lower limit. However this does happen sometimes on upper limit too.

[In this example](https://codesandbox.io/s/simple-composed-chart-forked-idtkzu) (modified from Recharts examples) domain lower limit is **1669240800000** and from tooltip you can see that first item on chart has value of **1669238914000**, this item should not be shown at all. Looks like there is some problem on domain/value compare functions.
![image](https://user-images.githubusercontent.com/42405685/206407722-ef6118f5-b508-4306-acf3-a442125bc1f9.png)

---

注意：不是用 https://recharts.github.io/recharts-issue-helper 创建的 issue 会被立即关闭。

**Comment by jukkahuuskonen** [comment]
Yes it is a bug. Read the line here (actually directly from your excerpt from docs)

> ...
> If set to true, **graphic elements (line, area, bars) will be clipped to conform to the specified domain**.

In my example graph shows a datapoint that is smaller than domain[0] and thus should not be shown in the chart.

This is especially needed feature when I have continous data (battery status, heart rate,...) that expands over the domain and I want to show line/area correctly within a subdomain of the full data. For some reason sometimes the chart is not cut properly and shows datapoints outside the given domain.

**Comment by ckifer** [comment]
Your example has `allowDataOverflow` set to true which will do just that, allow data to overflow past the domain. If you want it to be clipped, remove `allowDataOverlfow` and it will be clipped. 

I will add the bug label back to look into this more and confirm

**Comment by jukkahuuskonen** [comment]
> Your example has `allowDataOverflow` set to true which will do just that, allow data to overflow past the domain. If you want it to be clipped, remove `allowDataOverlfow` and it will be clipped.
> 
> I will add the bug label back to look into this more and confirm

I do not agree with this statement.

**allowDataOverflow** set to false will ignore the domain and show the whole data given to the chart.

**allowDataOverflow** set to true will allow the data to overflow outside the domain and clip lines, areas and bars to show just the part of the chart that is between the domain limits.

So I am using the **allowDataOverflow** correctly, but clipping the chart according to the given domain doesn't work 100% correctly.

See charts below:
On the top chart **allowDataOverflow =true** and the area is drown correctly continuing to the edge of the chart defined by domain[0], but for some reason there seems to be a datapoint show that is outside of the domain[0] (smaller than domain[0], which you can actually see on the chart in my original post. That first datapoint shouldn't be showing there because it is outside of the domain.

On the bottom chart **allowDataOverflow=false** and the whole dataset is shown in the chart ignoring the domain, exactly like the docs say:

> When domain of the axis is specified and the type of the axis is 'number', if allowDataOverflow is set to be false, the domain will be adjusted when the minimum value of data is smaller than domain[0] or the maximum value of data is greater than domain[1] so that the axis displays all data values.

![image](https://user-images.githubusercontent.com/42405685/207107180-4dfda572-44bc-40e7-ab56-ac558a1514d0.png)

**Comment by ckifer** [comment]
Ignore the above sorry, responding from my phone results in incorrect statements lol.

You're right - thanks for updating the example. Will look into it!

**Comment by jukkahuuskonen** [comment]
Anything happening with this?

**Comment by ckifer** [comment]
Feel free to look into it and contribute if you can find the source/a fix. This isn't currently being worked on.
